saas/axops/src/applatix.io/axops/configuration/configuration.go

```diff
@@ -133,3 +194,33 @@ func ProcessConfigurationStr(configStr string) (string, *axerror.AXError) {
 	}
 	return configValue, nil
 }
+
+func CreateKubernetesSecret(config *ConfigurationData) *axerror.AXError {
+	// Make sure log not printing out the secret content
+	utils.InfoLog.Println("[AXMON] Creating kube secret")
+	secret := map[string]interface{}{
+		"namespace": config.ConfigurationUser,
+		"name":      config.ConfigurationName,
+		"data":      config.ConfigurationValue,
+	}
+
+	axErr, _ := utils.AxmonCl.PostWithTimeRetry("secret", nil, secret, nil, retryConfig)
```

Comment:
i already have a retry in axmon. however it is a good idea to have retries if axmon itself is having trouble. Also, we need to see if the timeout you specify is longer than the retry period of axmon.